src/api/buildkite/annotate.ts

```diff
@@ -44,6 +49,18 @@ export const annotate = async (
     return;
   }
 
+  // Check if the annotation exceeds the maximum size
+  const buffer = Buffer.from(markdown, 'utf-8');
```

Comment:
I might be missing something obvious! What's the reason to use a buffer here?

Reply:
As opposed to `string.length()` or something else?

My understanding is that the buildkite annotation max is based on a byte size rather than a character length, so this is a more accurate comparison and would account for any characters that may be longer than a byte.

Reply:
Interesting, I'm not seeing the distinction drawn in https://buildkite.com/docs/agent/v3/cli-annotate

Reply:
Also, I think string length should be ok on that front:

```
$ node
> '🙂‍↔️' .length
5
```

Reply:
Oh wow! Did not know .length behaved like that.

Reply:
`String.length` counts UTF-16 code units though. Can we confirm how Buildkite measures its limit?